src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -21,520 +21,550 @@
-     * When the inventory is full, everything get's dumped into the building chest.
-     * But you can use this method to hold some stacks back.
-     *
-     * @param stack the stack to decide on
-     * @return true if the stack should remain in inventory
-     */
-    @Override
-    protected boolean neededForWorker(@Nullable final ItemStack stack)
-    {
-        return stack != null && Utils.isHoe(stack);
-    }
-
-    /**
-     * Returns the farmer's worker instance. Called from outside this class.
-     *
-     * @return citizen object
-     */
-    @Nullable
-    public EntityCitizen getCitizen()
-    {
-        return worker;
-    }
-}
+     }
+
+     /**
+      * Sets the hoe as held item.
+      */
+     private void equipHoe()
+     {
+         worker.setHeldItem(getHoeSlot());
+     }
+
+     /**
+      * Handles the offset of the field for the farmer.
+      *
+      * @param field the field object.
+      * @return true if successful.
+      */
+     private boolean handleOffset(@NotNull Field field)
+     {
+         if (workingOffset == null)
+         {
+             workingOffset = new BlockPos(field.getLengthPlusX(), 0, field.getWidthPlusZ());
+         }
+         else
+         {
+             if (workingOffset.getZ() <= -field.getWidthMinusZ() && workingOffset.getX() <= -field.getLengthMinusX())
+             {
+                 workingOffset = null;
+                 return false;
+             }
+             else if (workingOffset.getX() <= -field.getLengthMinusX())
+             {
+                 workingOffset = new BlockPos(field.getLengthPlusX(), 0, workingOffset.getZ() - 1);
+             }
+             else
+             {
+                 workingOffset = new BlockPos(workingOffset.getX() - 1, 0, workingOffset.getZ());
+             }
+         }
+         return true;
+     }
+
+     /**
+      * Resets the basic variables of the class.
+      */
+     private void resetVariables()
+     {
+         requestSeeds = true;
+         shouldDumpInventory = true;
+         shouldTryToGetSeed = true;
+     }
+
+     /**
+      * Get's the slot in which the hoe is in.
+      *
+      * @return slot number
+      */
+     private int getHoeSlot()
+     {
+         return InventoryUtils.getFirstSlotContainingTool(getInventory(), Utils.HOE);
+     }
+
+     /**
+      * Executes the planting of the field.
+      *
+      * @return the next state.
+      */
+     @NotNull
+     private AIState plant()
+     {
+         @Nullable BuildingFarmer buildingFarmer = getOwnBuilding();
+
+         if (buildingFarmer == null || buildingFarmer.getCurrentField() == null)
+         {
+             return AIState.PREPARING;
+         }
+         @Nullable Field field = getOwnBuilding().getCurrentField();
+
+         if (workingOffset != null)
+         {
+             BlockPos position = field.getLocation().down().south(workingOffset.getZ()).east(workingOffset.getX());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'position' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)